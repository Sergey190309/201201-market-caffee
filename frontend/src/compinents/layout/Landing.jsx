import React from 'react'
import {
  Segment,
  Container,
  Header
} from 'semantic-ui-react'

const Landing = () => (
    <Container>
      <Segment color='violet'>
        <Header
        textAlign='center'
        size='medium'
        color='yellow'>
          That's header
        </Header>
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Atque quis fugiat delectus repellat iste, nisi, temporibus ullam impedit officia nulla reprehenderit quas cupiditate libero sapiente enim illo veniam harum velit.
        </p>
        <p>
          Lorem ipsum dolor sit amet consectetur, adipisicing elit. Dolore necessitatibus maxime veritatis expedita quos fuga repudiandae. Eligendi, doloremque fugiat, similique, repellendus dolores velit magni ab quibusdam nesciunt eos accusantium reiciendis!
        </p>
      </Segment>
    </Container>
  )

export default Landing
