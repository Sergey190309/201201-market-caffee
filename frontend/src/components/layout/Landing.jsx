import React from "react";
import { Segment, Container, Header } from "semantic-ui-react";

const Landing = () => (
  <Container>
    <Segment color="violet">
      <Header textAlign="center" size="medium" color="yellow">
        That's header
      </Header>
      <p>Lorem ipsum dolor sit, amet consectetur adipisicing elit. Iusto, provident ratione numquam praesentium dolorem nostrum dolore asperiores in error dolores nihil quidem reprehenderit ut id maiores deleniti officiis eaque voluptates.</p>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Atque quis
        fugiat delectus repellat iste, nisi, temporibus ullam impedit officia
        nulla reprehenderit quas cupiditate libero sapiente enim illo veniam
        harum velit.
      </p>
      <p>Lorem ipsum dolor sit amet consectetur adipisicing elit. Quibusdam facere possimus eligendi delectus recusandae a quaerat, at officiis! Culpa odit sed adipisci, saepe eum repellat distinctio cupiditate tempora magni totam!</p>
    </Segment>
  </Container>
);

export default Landing;
