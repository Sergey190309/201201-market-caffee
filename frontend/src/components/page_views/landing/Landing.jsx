import isEmpty from 'lodash/isEmpty';
import React, { useState, useEffect } from 'react';
import { useSelector } from 'react-redux';
import PropTypes from 'prop-types';
import { Segment, Container } from 'semantic-ui-react';

import ElementSwitcher from '../view_elements/ElementSwitcher';
import { lngSelector, structureSelector } from '../../../redux/slices';

import { viewSegmentColor } from '../../../utils/colors';

export const getLoadedStructure = (pageName, structures) => {
  /**
   * The function created for testing.
   * Reciev all structures, return one that corresspong to component name
   */
  const { [pageName]: value } = structures;
  return value;
};

export const Landing = ({ getLoadedStructure }) => {
  const [structure, setStructure] = useState({});
  const [componentName] = useState('landing');
  const { lng } = useSelector(lngSelector);
  const { loaded } = useSelector(structureSelector);

  const loadedStructures = useSelector(structureSelector);

  useEffect(() => {
    if (loaded) {
      setStructure(getLoadedStructure(componentName, loadedStructures));
    }
  }, [loaded, getLoadedStructure, componentName, loadedStructures]);

  const _output = structure => {
    return <ElementSwitcher viewName={componentName} structure={structure} lng={lng} />;
  };
  // console.log('Landing, loadedStructure ->', loadedStructure)

  return (
    <Container data-testid='LandingContainer'>
      <Segment color={viewSegmentColor} data-testid='LandingSegment'>
        {/* {_output(structure)} */}
        {isEmpty(structure) ? null : _output(structure)}
      </Segment>
    </Container>
  );
};

Landing.defaultProps = {
  getLoadedStructure: getLoadedStructure,
};

Landing.propTypes = {
  getLoadedStructure: PropTypes.func.isRequired,
};

export default Landing;
