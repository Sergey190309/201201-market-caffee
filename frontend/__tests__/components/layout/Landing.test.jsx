import React from "react";
import renderer from "react-test-renderer";
import { cleanup } from "@testing-library/react";
import Landing from "../../../src/components/layout/Landing";

describe("Landing component", () => {
  test("snapshot", () => {
    const landing = renderer.create(<Landing />).toJSON();
    expect(landing).toMatchSnapshot();
  });
  afterAll(cleanup);
});
