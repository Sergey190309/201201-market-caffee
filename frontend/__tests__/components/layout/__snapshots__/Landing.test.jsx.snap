// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing component snapshot 1`] = `
<div
  className="ui container"
>
  <div
    className="ui violet segment"
  >
    <div
      className="ui yellow medium center aligned header"
    >
      That's header
    </div>
    <p>
      Lorem ipsum dolor sit, amet consectetur adipisicing elit. Iusto, provident ratione numquam praesentium dolorem nostrum dolore asperiores in error dolores nihil quidem reprehenderit ut id maiores deleniti officiis eaque voluptates.
    </p>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Atque quis fugiat delectus repellat iste, nisi, temporibus ullam impedit officia nulla reprehenderit quas cupiditate libero sapiente enim illo veniam harum velit.
    </p>
  </div>
</div>
`;
